xgcm/autogenerate.py

```diff
@@ -1,254 +1,240 @@
-        load class from xgcm.autogenerate) and coordinates (uses existing
-        xgc.grid logic)
-        """
-
-        #TODO: I can definitely infer the dims from the comodo stuff, but for
-        #now leave it explicit
-
-        for ki in axes.keys():
-            if ki not in dims.keys():
-                raise RuntimeError('dimension '+ki+' was not specified in axes \
-                                    input')
-            old_name = axes[ki]
-            dim = dims[ki]
-            new_name = old_name+'_inferred'
-
-            if ki == 'X':
-                periodic = True
-                boundary = None
-                fill_value = 0.0
-                wrap = x_wrap
-            elif ki == 'Y':
-                periodic = True
-                boundary = None
-                fill_value = 0.0
-                wrap = y_wrap
-            elif ki == 'Z':
-                periodic = False
-                boundary = 'fill'
-                # For depth decide between surface or bottom fill
-                if z_pad is None:
-                    # Infer limits from depth data
-                    z_min = ds[old_name].min().data
-                    z_max = ds[old_name].max().data
-                    # The value below is an attempt on extrapolating the values
-                    # linearly. The difference between values at the top and
-                    # bottom is used to pad the depth array at the appropriate
-                    # boundary. For multidimensional arrays the min difference
-                    # is chosen. This could lead to undesired results if the
-                    # depth spacing is spatially irregular (though thats a rare
-                    # case IMO)
-                    top_diff = ds[old_name].diff(dim).isel(**{dim: 0}).min()
-                    bot_diff = ds[old_name].diff(dim).isel(**{dim: -1}).min()
-                    top = z_min.data - top_diff.data
-                    bot = z_max.data + bot_diff.data
-                    if to is 'left':
-                        fill_value = top
-                    elif to is 'right':
-                        fill_value = bot
-                else:
-                    fill_value = z_pad
-
-            # Input coordinate has to be declared as center,
-            # or xgcm.Axis throws error. Will be rewrapped below.
-            attrs = ds[old_name].attrs
-            attrs['axis'] = ki
-            attrs.pop('c_grid_axis_shift', None)
-            ds[old_name].attrs = attrs
-
-            if dim_switch:
-                ax = AutoGenerate(ds, ki, periodic=periodic, wrap=wrap)
-                ds.coords[new_name] = ax._generate_coord(ds[old_name], to,
-                                                         boundary=boundary,
-                                                         fill_value=fill_value)
+from .grid import Axis
+
+
+def generate_axis(ds,
+                  axis,
+                  name,
+                  axis_dim,
```

Comment:
The docstring is not consistent with the function arguments

Reply:
Fixed it.